src/components/utils/_functions.ts

```diff
@@ -0,0 +1,21 @@
+import postsFromServer from '../../api/posts';
+import commentsFromServer from '../../api/comments';
+import usersFromServer from '../../api/users';
+
+const GetCommentsById = (postId: number) => (
+  commentsFromServer.filter(comment => (
+    comment.postId === postId
+  ))
+);
+
+const getUserById = (id: number) => (
+  usersFromServer.find((user) => (
+    user.id === id
+  ))
+);
+
+export const posts = postsFromServer.map(post => ({
+  ...post,
+  user: getUserById(post.userId),
```

Comment:
it's better to use `null` instead of `undefined`
if getUserById returns undefined write null in the user key